### PR TITLE
Wait for the kernel module to load

### PR DESCRIPTION
**Pull request by rhatdan**
There is a race condition possible, where we load the overlay kernel module
but it does not complete before we check to see if it loaded.  This patch
causes to code to check three times with a 1 second sleep between checks.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
The goal of this pull request is to fix this issue.  https://github.com/containers/storage/issues/84

It is a sucky fix, but not sure what we could do better.

File: drivers/overlay/overlay.go [hunk 1]
```diff
@@ -281,10 +278,26 @@ func supportsOverlay() error {
...
+	var err error
+	// We can try to modprobe overlay first before looking at
+	// proc/filesystems for when overlay is supported
+	exec.Command("modprobe", "overlay").Run()
+	for i := 0; i < 3; i++ {
+		if err = readOverlayFilesystem(); err == nil {
+			return nil
+		}
+		if i < 2 {
+			time.Sleep(1 * time.Second)
```
Comment (TomSweeneyRedHat) [inline drivers/overlay/overlay.go]: should we retry if we get the graphdriver.ErrNoSupported error from line 281?

File: drivers/overlay/overlay.go [hunk 2]
```diff
@@ -281,10 +278,26 @@ func supportsOverlay() error {
...
-	logrus.Error("'overlay' not found as a supported filesystem on this host. Please ensure kernel is new enough and has overlay support loaded.")
 	return errors.Wrap(graphdriver.ErrNotSupported, "'overlay' not found as a supported filesystem on this host. Please ensure kernel is new enough and has overlay support loaded.")
 }
 
+func supportsOverlay() error {
+	var err error
+	// We can try to modprobe overlay first before looking at
+	// proc/filesystems for when overlay is supported
+	exec.Command("modprobe", "overlay").Run()
+	for i := 0; i < 3; i++ {
```
Comment (TomSweeneyRedHat) [inline drivers/overlay/overlay.go]: Might be good to leave a comment here on the race condition and the thinking for the retry.

**Comment by TomSweeneyRedHat** [comment]
couple comments/questions.  Travis isn't happy.

**Comment by rhatdan** [comment]
@nalind questions if this is actually a race condition, so I will close along with the issue, as a Works For Me.  Probably would only happen one time on a fresh boot, so tough to create.

**Pull request closed without merging**